Remove duplicated build system in alien/ArcaneInterface

guignont opened the pull request:
A proposal to remove buildsystem part in Alien ArcaneInterface

guignont committed:
- use alien with build system in app_buildsystem_tools
- remove obsoletes configs

guignont committed:
- make all cmake directory deprecated in ArcaneInterface

guignont committed:
Merge branch 'main' into tg-clean-alien-buildsystem

guignont committed:
- remove cmake deprecated

guignont committed:
- remove cmake deprecated

guignont committed:
- wip

grospelliergilles reviewed (commented):
I think it is a good idea to remove duplication.
So now it uses the configuration from `app_buildsystem_tools` ?

guignont commented:
> I think it is a good idea to remove duplication. So now it uses the configuration from `app_buildsystem_tools` ?

Yes

guignont committed:
- just for test

guignont committed:
- just for test

guignont commented:
@grospelliergilles  It's seem that Ubuntu used for Codecov test lacks Mono and DotNet.
Is it possible to install them ?

grospelliergilles commented:
> @grospelliergilles It's seem that Ubuntu used for Codecov test lacks Mono and DotNet. Is it possible to install them ?

we no longer support `mono` but `dotnet` is present in the image used by codecov. You can see in this action for example; https://github.com/arcaneframework/framework/actions/runs/12191621972/job/34010826688. The tests used by `dotnet` contains the string `corecrl` in their name.

guignont committed:
- since AlienPlugins do not require Mono we remove it as essential package.

guignont committed:
- FindDotNet.cmake try to look at mono compiler but overrides it at the end

guignont committed:
- since XBUILD with dotnet overrides XBUILD with mono simplify search for dotnet binary
- also XBUILD_ARGS are deleted at the end so remove all settings related to XBUILDS_ARGS

stdcm reviewed (changes requested):
Thanks for the fixes. The Arcane CI is ok but when installing and trying with ArcaneDemo, AlienLegacyPlugins is not found. Maybe the modifications to handle AlienPlugins are not fully reported in app_buildsystem_tools...?

guignont commented:
@stdcm 
Can you describe framework build and install procedure for use with ArcaneDemo because my ArcaneDemo build has no problem finding  AlienLegacyPlugins ?

guignont committed:
Merge branch 'main' into tg-clean-alien-buildsystem

guignont commented:
I just made the following test in ArcaneDemo replacing build-system with the one in this branch.
The configuration with cmake works fine but when compiling there is errors in c++ generated files:

In file included from /work/guignont/work/ArcaneDemo/cmake-build-debug-arcane3user_gimkl2018b_localarcane/ArcaneDemo/axl/RootTester_axl.h:507:0,
                 from /work/guignont/work/ArcaneDemo/common/ArcGeoSim/src/ArcGeoSim/Tests/ServiceTesters/StrongOptionsTester/RootTesterService.h:10,
                 from /work/guignont/work/ArcaneDemo/common/ArcGeoSim/src/ArcGeoSim/Tests/ServiceTesters/StrongOptionsTester/RootTesterService.cc:6:
/work/guignont/work/ArcaneDemo/cmake-build-debug-arcane3user_gimkl2018b_localarcane/ArcaneDemo/axl/RootTester_CaseOptionsT.h:133:65: error: « ; » attendu à la fin de la déclaration du membre
   Arcane::ConstArrayView<TypesStrongOptionsTester::eSimpleEnum> simpleEnum4 const
                                                                 ^~~~~~~~~~~
/work/guignont/work/ArcaneDemo/cmake-build-debug-arcane3user_gimkl2018b_localarcane/ArcaneDemo/axl/RootTester_CaseOptionsT.h:134:3: error: expected unqualified-id before « { » token
   {
   ^
/work/guignont/work/ArcaneDemo/cmake-build-debug-arcane3user_gimkl2018b_localarcane/ArcaneDemo/axl/RootTester_CaseOptionsT.h:137:65: error: « ; » attendu à la fin de la déclaration du membre
   Arcane::ConstArrayView<TypesStrongOptionsTester::eSimpleEnum> simpleEnum5 const
                                                                 ^~~~~~~~~~~
/work/guignont/work/ArcaneDemo/cmake-build-debug-arcane3user_gimkl2018b_localarcane/ArcaneDemo/axl/RootTester_CaseOptionsT.h:138:3: error: expected unqualified-id before « { » token
   {

Anyway with a manual correction the build is ok.

stdcm commented:
The work initiated here has been concluded in others PRs now merged in main branch. Closing this PR.
